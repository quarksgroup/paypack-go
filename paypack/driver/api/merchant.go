package api

import (
	"context"

	"github.com/quarksgroup/paypack-go/paypack"
)

type merchantService struct {
	http *wrapper
}

func (s *merchantService) Me(ctx context.Context) (*paypack.Merchant, error) {

	const endpoint = "merchants/me"

	out := new(merchantResponse)

	_, err := s.http.do(ctx, "GET", endpoint, nil, out)

	res := &paypack.Merchant{
		Name:    out.Name,
		InRate:  out.In,
		OutRate: out.Out,
		Balance: out.Balance,
	}

	return res, err
}

var _ (paypack.MerchantService) = (*merchantService)(nil)
