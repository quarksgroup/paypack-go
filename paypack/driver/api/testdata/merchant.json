{
    "In_rate": 0.023,
    "out_rate": 0.023,
    "name": "john doe",
    "balance": 100
}