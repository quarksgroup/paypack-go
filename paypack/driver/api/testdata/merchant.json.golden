{
    "id": "SFDHSK",
    "name": "john doe",
    "in_rate": 0.023,
    "out_rate": 0.023,
    "balance": 100
}